core/modules/syncer.js

```diff
@@ -33,6 +33,15 @@ function Syncer(options) {
 	var self = this;
 	this.wiki = options.wiki;
 	this.syncadaptor = options.syncadaptor;
+	if (this.syncadaptor && this.syncadaptor.addEventListener) {
```

Comment:
Why do we make the sync adaptor be an event listener? Wouldn't it make more sense for the syncer to be an event listener?

I don't actually think it makes much sense to use the event pattern here, because the syncer and sync adaptor are tightly integrated components. Better to for the syncer to pass a reference to itself to the syncadaptor constructor, and then have the syncadaptor invoke methods on it.

Reply:
In the actual startup code, both are initialized by the "startup" startup module, and the syncadaptor is passed to the syncer constructor. The syncer constructor would have to then call a method on the syncadaptor with `this` as an argument. If that's fine, then we're good.